dagfactory/utils.py

```diff
@@ -426,3 +427,21 @@ def cast_with_type(data):
         return [cast_with_type(item) for item in data]
 
     return data
+
+
+def merge_dict(dict1: dict, dict2: dict) -> dict:
```

Comment:
I believe we should avoid extending the definitions in the `utils.py` module. My vote is usually not not have this type of module. It tends to grow, and it is hard to track its contents.

I believe we should rename this method ot `_merge_dicts`, define ti inside the `DagFactory` class and join the implementation done in 
https://github.com/astronomer/dag-factory/blob/2fd9c757d564eed1686f24edf353d166db518d64/dagfactory/dagfactory.py#L138-L163

With this one.

The advantage of this one is that it handles generically merges of values that map to lists and dictionaries (the other two functions were specific to certain keys).

The advantage of the other `_merge` methods is that they handle a list of dictionaries. We should combine both to get the best combination.